### ISSUE TITLE
`github-annotation-native` annotation is limited to 10 annotations.

### ISSUE THREAD

**Issue by stryder78** [sqlfluff/sqlfluff#3857]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In Github Actions I created a workflow that uses `sqlfluff lint --format github-annotation` and `sqlfluff lint --format github-annotation-native`. But `github-annotation-native` does not annotate all linting violations.  

For example, in screenshots below, it only makes 10 annotations vs ~500 when using `github-annotation` linting the same query. The strange part is it seems to annotate properly when there is <= 10 linting violations but cuts off when > 10.

### Expected Behaviour

Expected `github-annotation-native` option to annotate the entire PR like `github-annotation`.

`github-annotation` result after running CI - 545 annotations:
![github-annotation output](https://user-images.githubusercontent.com/89415467/191286373-c852c30d-9644-4589-98f8-edbd351154d4.png)

### Observed Behaviour

On a .sql file with >10 linting violations `github-annotation-native` does not annotate all linting violations in pull request.

`github-annotation-native` result after running CI when linting same query - only 10 annotations expected 545 like above:
![github-annotation-native output](https://user-images.githubusercontent.com/89415467/191286862-d5f5d88d-53da-4ac1-a49d-e92954a66b3d.png)

### How to reproduce

`ci.yaml` when using `github-annotation`:
```
      # lint sql
      - name: 🧪 Lint dbt models
        id: sqlfluff_json
        if: steps.changed-files.outputs.any_changed == 'true'
        shell: bash -l {0}
        run: |
          sqlfluff lint --format github-annotation --annotation-level error --nofail ${{ steps.changed-files.outputs.all_changed_files }} > ./annotations.json

      # annotate PR
      - name: Annotate PR
        uses: yuzutech/annotations-action@v0.4.0
        with:
          repo-token: "${{ secrets.GITHUB_TOKEN }}"
          title: "SQLFluff Lint"
          input: "./annotations.json"
```
`ci.yaml` when using `github-annotation-native`:
```
      # lint sql
      - name: 🧪 Lint dbt models
        id: sqlfluff_json
        if: steps.changed-files.outputs.any_changed == 'true'
        shell: bash -l {0}
        run: |
          sqlfluff lint --format github-annotation-native --annotation-level error --nofail ${{ steps.changed-files.outputs.all_changed_files }}

```

### Dialect

snowflake

### Version

sqlfluff, version 1.3.0

### Configuration

.sqlfluff:
```
[sqlfluff]
dialect = snowflake
templater = dbt
exclude_rules = L031, L011, L001

[sqlfluff:templater:dbt]
project_dir = ./
profiles_dir = ./conf_sqlfluff
profile = default

[sqlfluff:rules:L016]
max_line_length = 999

[sqlfluff:rules:L011]
aliasing = implicit
```
.sqlfluffignore:
```
target/
dbt_modules/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
Can you provide a SQL test file? You may be right about the "10+ violations" thing, but maintainers are more wiling to look at an issue with a repro case.

**Comment by Nazaniiin** [comment]
I am seeing the same behavior when using `github-annotation-native`. 

With this flag, only the first 10 violations are annotated even though I see more than 10 violations in the output.

## Screenshot of part of the output

![257303871-b84eff04-017b-4ab5-9d69-6ae4593432fb](https://github.com/sqlfluff/sqlfluff/assets/8875071/13ed06ed-cf07-4214-a677-49426f8d5934)


## Dialect

Snowflake

## Version

`2.1.3`

## Configuration

`.sqlfluff`

```bash
[sqlfluff]
templater = dbt
dialect = snowflake
warnings = RF04
max_line_length = 80
sql_file_exts = .sql
processes = 1

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:dbt]
project_dir = .
profile = default
target = sqlfluff
```

`.pre-commit-config.yml`

```yml
- repo: https://github.com/sqlfluff/sqlfluff
    rev: 2.1.3
    hooks:
      - id: sqlfluff-lint
        stages: [manual]
        description: "Returns linting errors with `SQLFluff`"
        args: [
          "--config", "dbt/.sqlfluff",
          "--ignore-local-config", "dbt/.sqlfluffignore",
          "--format", "github-annotation-native",
          "--annotation-level", "failure",
        ]
        additional_dependencies: [
          'dbt-snowflake==1.5.0',
          'sqlfluff-templater-dbt==2.1.3'
        ]
```

Step for running the check in GitHub Actions:

```yml
- name: Get new and changed .sql files in DBT project
   id: get_files_to_lint
   run: |
      echo "ACMD_FILES=$(
      echo \
      $(echo ${{ steps.changed_files.outputs.all_changed_files }} |
      tr -s ' ' '\n' |
      grep -E '^dbt.*[.]sql$' |
      tr -s '\n' ' ')
      )" >> $GITHUB_OUTPUT
- name: Run SQLFluff linter on changed files
   if: steps.get_files_to_lint.outputs.ACMD_FILES != ''
   run: |
     pre-commit run --hook-stage manual sqlfluff-lint --files ${{ steps.get_files_to_lint.outputs.ACMD_FILES }} --verbose
```

**Comment by alanmcruickshank** [comment]
Hmmmmmm _interesting_. I've confirmed this bug with this PR: #5449 .

Looking at the annotations (here: https://github.com/sqlfluff/sqlfluff/pull/5449/files#annotation_16095232428), only the first 10 _exactly_ are shown in the native process, but all are shown using the yuzutech action.

**Comment by alanmcruickshank** [comment]
Ok - frustratingly, this appears to be a GitHub limitation rather than a SQLFluff one. On digging around the internet, I've found numerous people complaining about this:
- https://github.com/orgs/community/discussions/26680
- https://www.reddit.com/r/github/comments/122rmw7/any_way_to_get_around_the_annotation_limitation/
- https://github.com/orgs/community/discussions/68471
- https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md

I think at this stage, all we can do is recommend that people use the other option in the docs and provide links to these articles.
